lib/Commands/BuildEngineCommand.cpp

```diff
@@ -104,7 +104,7 @@ struct AckermannKey {
   /// Convert an Ackermann key to its encoded representation.
   operator core::KeyType() const {
     char inputKey[32];
-    sprintf(inputKey, "ack(%d,%d)", m, n);
+    snprintf(inputKey, sizeof(inputKey), "ack(%d,%d)", m, n);
```

Comment:
We should use the statically known size here (and everywhere else) instead of `sizeof`.